crates/cdk/src/nuts/nut17/mod.rs

```diff
@@ -137,13 +145,27 @@ impl AsRef<SubId> for Params {
     }
 }
 
-impl From<Params> for Vec<Index<(String, Kind)>> {
+impl From<Params> for Vec<Index<Notification>> {
     fn from(val: Params) -> Self {
         let sub_id: SubscriptionGlobalId = Default::default();
         val.filters
-            .iter()
-            .map(|filter| Index::from(((filter.clone(), val.kind), val.id.clone(), sub_id)))
-            .collect()
+            .into_iter()
+            .map(|filter| {
+                let idx = match val.kind {
+                    Kind::Bolt11MeltQuote => {
+                        Notification::MeltQuoteBolt11(Uuid::from_str(&filter)?)
+                    }
+                    Kind::Bolt11MintQuote => {
+                        Notification::MintQuoteBolt11(Uuid::from_str(&filter)?)
+                    }
+                    Kind::ProofState => Notification::ProofState(PublicKey::from_str(&filter)?),
+                };
+
+                Ok(Index::from((idx, val.id.clone(), sub_id)))
+            })
+            .collect::<Result<_, anyhow::Error>>()
+            .unwrap()
+        // TODO don't unwrap, move to try from
```

Comment:
@crodas I think we should merge this and then rebase your PR https://github.com/cashubtc/cdk/pull/473 on this. Could you make this a `TryFrom` there is it still relevant I haven't reviewed that PR yet but know it changes a good bit in these files. What do you think?

Reply:
Can I keep the `unwrap` into the code for new?

I think moving to try_from will imply some changes in our custom pubsub lib, which may require rethinking part it's design, which I don't have context about and is slightly out of the scope of this PR

Reply:
Yes lets do that separately

Reply:
> We should not unwrap anything; we can either bubble up the error or exit logging it if it is unexpected (a malformed user-given input is not an error). Otherwise, it could be an attack vector that reboots the mint to flood the log with garage entries.

I agree we should not have unwraps in the codebase. Its not a question of do we keep the unwrap in the codebase its a question of do we fix this here in this PR or in your immediate follow up PR related to wallet WS #473. As changing it here may cause a lot of merge conflicts for you. 

> It shouldn't be enforced unless there is a standard/NUT that enforces quotes ID to be UUIDs.

This is why we don't and can't enforce it for the wallet. However for the mint it is an implementation choice that we are free to make and it has always been UUID in cdk just stored as a string and treated as a string so there are no legacy IDs. 

> Having a Uuid is fine, but it shouldn't be enforced, instead hinted, as it should be backward compatible. Perhaps the ID is an enum of Uuid(Uuid), Legacy(String). 

I think using an enum makes it more complicated and gives flexibility we don't need to the mint as we would always use the Uuid variant of the enum. I believe it better to take an opinionated stance for the reasons @tdelabro pointed to in the description. Supporting backwards compatibility should only be done in cases where it is absolutely needed as it leads to significant tech debt that should be avoided where possible.